code/game/objects/items/weapons/RCD.dm

```diff
@@ -11,312 +11,180 @@ RCD
-	if (acc == "all")
-		conf_access = null
-	else if(acc == "one")
-		use_one_access = !use_one_access
-	else
-		var/req = text2num(acc)
-
-		if (conf_access == null)
-			conf_access = list()
-
-		if (!(req in conf_access))
-			conf_access += req
-		else
-			conf_access -= req
-			if (!conf_access.len)
-				conf_access = null
-
-/obj/item/weapon/rcd/verb/change_airlock_setting()
-	set name = "Change Airlock Setting"
-	set category = "Object"
-	set src in usr
-
-	var airlockcat = input(usr, "Select whether the airlock is solid or glass.") in list("Solid", "Glass")
-	switch(airlockcat)
-		if("Solid")
-			if(advanced_airlock_setting == 1)
-				var airlockpaint = input(usr, "Select the paintjob of the airlock.") in list("Default", "Engineering", "Atmospherics", "Security", "Command", "Medical", "Research", "Mining", "Maintenance", "External", "High Security")
-				switch(airlockpaint)
-					if("Default")
-						airlock_type = /obj/machinery/door/airlock
-					if("Engineering")
-						airlock_type = /obj/machinery/door/airlock/engineering
-					if("Atmospherics")
-						airlock_type = /obj/machinery/door/airlock/atmos
-					if("Security")
-						airlock_type = /obj/machinery/door/airlock/security
-					if("Command")
-						airlock_type = /obj/machinery/door/airlock/command
-					if("Medical")
-						airlock_type = /obj/machinery/door/airlock/medical
-					if("Research")
-						airlock_type = /obj/machinery/door/airlock/research
-					if("Mining")
-						airlock_type = /obj/machinery/door/airlock/mining
-					if("Maintenance")
-						airlock_type = /obj/machinery/door/airlock/maintenance
-					if("External")
-						airlock_type = /obj/machinery/door/airlock/external
-					if("High Security")
-						airlock_type = /obj/machinery/door/airlock/highsecurity
-			else
-				airlock_type = /obj/machinery/door/airlock
-
-		if("Glass")
-			if(advanced_airlock_setting == 1)
-				var airlockpaint = input(usr, "Select the paintjob of the airlock.") in list("Default", "Engineering", "Atmospherics", "Security", "Command", "Medical", "Research", "Mining")
-				switch(airlockpaint)
-					if("Default")
-						airlock_type = /obj/machinery/door/airlock/glass
-					if("Engineering")
-						airlock_type = /obj/machinery/door/airlock/glass_engineering
-					if("Atmospherics")
-						airlock_type = /obj/machinery/door/airlock/glass_atmos
-					if("Security")
-						airlock_type = /obj/machinery/door/airlock/glass_security
-					if("Command")
-						airlock_type = /obj/machinery/door/airlock/glass_command
-					if("Medical")
-						airlock_type = /obj/machinery/door/airlock/glass_medical
-					if("Research")
-						airlock_type = /obj/machinery/door/airlock/glass_research
-					if("Mining")
-						airlock_type = /obj/machinery/door/airlock/glass_mining
-			else
-				airlock_type = /obj/machinery/door/airlock/glass
-		else
-			airlock_type = /obj/machinery/door/airlock
-
+	var/static/list/allowed_door_types = list(/obj/machinery/door/airlock = "Standard",
```

Comment:
I dunno why the list of available doors would change mid game. Considering a restart would be required.